### PR TITLE
track2 sdk:add client for snapshot,pipprefix, bump apiversion and upgrade network package

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
add client for snapshot,pipprefix, bump apiversion and upgrade network package
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by lodrem — APPROVED** [review]
LGTM

File: pkg/azclient/Makefile [hunk 1]
```diff
@@ -73,21 +73,23 @@ generate: install-dependencies build generatecode generateimpl vet-all
 .PHONY: generatecode
 generatecode: build ## Generate client
 	$(TYPESCAFFOLD) --package github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice/v2 --package-alias armcontainerservice --resource ManagedCluster --client-name ManagedClustersClient --apiversion 2021-10-01
-	$(TYPESCAFFOLD) --package github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2 --package-alias network --resource PrivateLinkService --client-name PrivateLinkServicesClient --verbs get,createorupdate,delete,list --apiversion 2021-08-01 --expand
 	$(TYPESCAFFOLD) --package github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources --package-alias resources --resource Deployment --client-name DeploymentsClient --verbs delete --apiversion 2017-05-10
-	$(TYPESCAFFOLD) --package github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4 --package-alias armcompute --resource Disk --client-name DisksClient --apiversion 2022-03-02
+	$(TYPESCAFFOLD) --package github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4 --package-alias armcompute --resource Disk --client-name DisksClient --apiversion 2022-07-02
```
Comment (feiskyer) [inline pkg/azclient/Makefile:77]: are we able to reuse the same API Version from the SDK directly?

Comment (MartinForReal) [reply pkg/azclient/Makefile:77]: yes. I'll raise another pr

**Pull request merged**